os/fs/smartfs/smartfs_utils.c

```diff
@@ -2096,20 +2096,19 @@ ssize_t smartfs_append_data(FAR struct smartfs_mountpt_s *fs, FAR struct smartfs
 	struct smartfs_chain_header_s *chainheader;
 
 	while (buflen > 0) {
-		/* We will fill up the current sector. Write data to
-		 * the current sector first.
-		 */
+		/* Write data to the current sector as long as there is more space available.*/
 		fvdbg("Datalen : %d, sector : %d, offset : %d\nFilepos : %d, bytes written : %d, buflen : %d\n", sf->entry.datalen, sf->currsector, sf->curroffset, sf->filepos, byteswritten, buflen);
 
 #ifdef CONFIG_SMARTFS_USE_SECTOR_BUFFER
 		readwrite.count = fs->fs_llformat.availbytes - sf->curroffset;
-		if (readwrite.count > buflen) {
-			readwrite.count = buflen;
-		}
-
-		memcpy(&sf->buffer[sf->curroffset], &buffer[byteswritten], readwrite.count);
-		sf->bflags |= SMARTFS_BFLAG_DIRTY;
+		if (readwrite.count > 0) {
+			if (readwrite.count > buflen) {
```

Comment:
buflen is always greater than 0 here, so first if is unnecessary.

Reply:
I observed that it is not always like that.
Consider the case of file creation with a buffer size = 1013 i.e. available data bytes
We always fill up the sector completely, then during the next write, readwrite.count = 0 first
Then we sync this sector and fill up the next sector completely and the cycle repeats.